charters/rust-core/README.md

```diff
@@ -0,0 +1,277 @@
+# Rust Core Working Group Charter <!-- omit in toc -->
+
+**Start Date**: June 6th 2024
+
+**End Date**: Unknown
```

Comment:
I would recommend being fairly aggressive with this date. Reason being:

1. It forces us to re-consider the scope of work and whether we're on the right track or need to make adjustments within a given timeframe
2. It lights a 🔥 under our 🍑 to get some work done
3. We can always re-charter to extend the date and/or make other changes

It's also fine to have a smaller set of deliverables in this charter, but call out future deliverables: like if there's a deliverable we know will be done in 3 months and another that will take 1+ year we can keep the second deliverable out of scope, or narrow the scope to something like a 'prototype'.

I generally think a charter should be under a year in length, probably 2 quarters or so, which gives us enough time to plan and commit, but not too much for things to dramatically shift.

Reply:
Appreciate the comment @decentralgabe! Going to work on adding timelines here shortly. 

Deliverables list seems long here but I think it's necessary. We could remove the Swift deliverables but that seems ill-advised because we want to showcase the power of cross-language bindings for more than one language, else we may naturally build matters proprietary to the single target language.